### ISSUE TITLE
Hard Negatives Images

### ISSUE THREAD

**Issue by konstantin-frolov** [rwightman/efficientdet-pytorch#71]
Hi. I am trying to use your implementation of the model for detection in medical applications.
I am working with large images around 3000x4000. Then I use a sliding window over the large image and predict on the small image (1024, 1024).
I created a coco-like dataset only with interesting areas with objects detected (tumors, mass, etc.). But in the big image, I have another object in the form of text. And the network is activated at these sites.
I had the same problem when I used Mask RCNN, but fixing the code on the FPN network solves the problem. Now I can use images with blank coco annotations for training process. I call these images "Hard Negatives".
Can I use blank annotations in your implementation? Perhaps you can suggest any fixes for using it.

**Comment by lgvaz** [comment]
Let me quote Ross answer in one of my posts [here](https://github.com/rwightman/efficientdet-pytorch/issues/50):

> 'empty annotations' would be cls and box targets all zero, they are fixed-size tensors when passed to the model

I'm just not entirely sure if by "all zero" he means:
```python
bbox_targets = torch.zeros((0, 4))
```
Or
```python
bbox_targets = torch.zeros((1, 4))
```

**Comment by rwightman** [comment]
Ignoring the batch dimension here, you definitely want `bbox_targets = torch.zeros((N, 4))` where N > 0 at the bench/model inputs ... and you need to match the same N for the classes target. The way these models are normally used by me (and what I observed for similar setups) is the dimensions of bbox targets and classes remains the same from batch to batch

In my code bbox targets is `bbox_targets = torch.zeros((100, 4))` regardless of the number of bbox in the dataset, they start as zeros and valid entries get set if they exist. You don't change the dim from sample to sample. It may look like it is set to zeros(0, 4) based on my dataset, but it's my collate fn that fills in the final target arrays.

So the default for an image with no annotations would be all zeros for both bbox and class targets. As for how this impacts the loss for a negative image, I've yet to do any back of envelope analysis for the current bbox and class loss fn to determine what the gradients are if you had a 'negative image' with no annotations
